src/uct/tcp/tcp_ep.c

```diff
@@ -834,11 +834,65 @@ static inline void uct_tcp_ep_am_send(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
 
     uct_tcp_ep_send(ep);
 
-    if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
-        uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVWRITE, 0);
-    } else {
+    if (!uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
         uct_tcp_ep_ctx_reset(&ep->tx);
+    } else {
+        uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVWRITE, 0);
+    }
+}
+
+static inline void
+uct_tcp_ep_am_short_fill_data(uct_tcp_am_hdr_t *hdr, uint64_t header,
+                              const void *payload, unsigned length)
+{
+    *((uint64_t*)(hdr + 1)) = header;
+    memcpy((uint8_t*)(hdr + 1) + sizeof(header), payload, length);
```

Comment:
may be UCS_PTR_BYTE_OFFSET((hdr + 1), sizeof(header)) ?

@yosefe I seen same combination many times in send functions, what about add macro/inline function to unify header/payload initialization? (not in this PR :) )

Reply:
yep, sounds good

Reply:
> yep, sounds good

@yosefe @hoopoepg let's do it in the new PR?

Reply:
yes